src/software/amazon/ion/impl/IonReaderTextSystemX.java

```diff
@@ -171,10 +287,10 @@ else if (token_type == IonTokenConstsX.TOKEN_BINARY) {
         case IonTokenConstsX.TOKEN_UNKNOWN_NUMERIC:
             switch (_value_type) {
             case INT:
-                if (len <= MAX_IMAGE_LENGTH_INT) {
+                if (Radix.DECIMAL.isInt(s, len)) {
                     _v.setValue(Integer.parseInt(s));
                 }
-                else if (len <= MAX_IMAGE_LENGTH_LONG) {
+                else if (Radix.DECIMAL.isLong(s, len)) {
```

Comment:
This seems to be doing a lot more work than necessary.  What was wrong with the older, simpler logic?